apache2/waf_logging/waf_log_util.cc

```diff
@@ -91,77 +52,59 @@ void set_waf_format(waf_format::Waf_Format* waf_format, char* resourceId, char*
 				}
 			}
 			mandatory_message[ind + i] = '\0';
-			properties->set_message(mandatory_message);
+			message_str = mandatory_message;
 		}
 		else {
 			messages[strlen(messages) - 1] = '\0';
-			properties->set_message(messages+1);
+			message_str = messages+1;
 		}		
 	}
 
 	if (site == 0) {
-        properties->set_site(waf_format::Properties::Global);
+        site_str = "Global";
     }
-    
-    if (details_messages != NULL) {
-        details->set_message(details_messages);
+
+    if (details_file != NULL) {
+		details_file[strlen(details_file) - 1] = '\0';
     }
-    
+
     if (details_data != NULL) {
 		details_data[strlen(details_data) - 1] = '\0';
-        details->set_data(details_data+1);
-    }
-    
-    if (details_file != NULL) {
-        details->set_file(details_file);
     }
     
     if (details_line != NULL) {
 		details_line[strlen(details_line) - 1] = '\0';
-        details->set_line(details_line+1);
     }
-    
-	if (hostname != NULL) {
-		properties->set_hostname(hostname);
-	}
 
 	if (waf_unique_id != NULL) {
 		waf_unique_id[strlen(waf_unique_id) - 1] = '\0';
-		properties->set_transactionid(waf_unique_id+1);
 	}
+
+    waf_logging::waf_log waf_log{resourceId, operationName, category, waf_logging::property{ instanceId, clientIP, clientPort, requestUri, ruleSetType, ruleSetVersion, ruleId+1, message_str, action_str, site_str, waf_logging::detail{ details_messages, details_data+1, details_file+1, details_line+1 }, hostname, waf_unique_id+1}};
+    return waf_log;
 }
 
 // Main function:  get fields from modsec, set the protobuf object and write to file in json.
 int generate_json(char** result_json, char* resourceId, char* operationName, char* category, char* instanceId, char* clientIP, char* clientPort, const char* requestUri, char* ruleSetType, char* ruleSetVersion, char* ruleId, char* messages, int action, int site, char* details_messages, char* details_data, char* details_file, char* details_line, const char* hostname, char* waf_unique_id) {
-    waf_format::Waf_Format waf_format;
     string json_string;
-    google::protobuf::util::JsonPrintOptions options;
-    google::protobuf::util::Status convert_result;
     char* json_str;
+		waf_logging::waf_log waf_log;
```

Comment:
1) Let's move this inside the "try{" block below to reduce the scope of the object
2) Please made sure indentation is well-aligned, here it is broken